workflows/promote-job.ts

```diff
@@ -0,0 +1,103 @@
+import fs from 'fs-extra';
+import {cyan, green, red} from 'kleur';
+import {createPullRequest} from 'octokit-plugin-create-pull-request';
+import {Octokit as BaseOctokit} from '@octokit/rest';
+
+const Octokit = BaseOctokit.plugin(createPullRequest);
+const octokit = new Octokit({auth: process.env.GITHUB_TOKEN});
+
+const versionsJsonFile = 'configs/versions.json';
+const params = {owner: 'ampproject', repo: 'cdn-configuration'};
+
+// TODO(danielrozenberg): change to @ampproject/release-on-duty after testing is done.
+const releaseOnDuty = '@ampproject/wg-infra';
+
+type Versions = Record<string, string | null>;
+type CreatePullRequestResponsePromise = ReturnType<
+  typeof octokit.createPullRequest
+>;
+
+interface VersionMutatorDef {
+  versionsChanges: Versions;
+  title: string;
+  body: string;
+}
+
+function timeExecutionTime_(startTime: number): string {
```

Comment:
I don't think we need timing when it's available in the GH workflow

Reply:
Done fbf646c